docs/DeveloperGuide.md

```diff
@@ -296,6 +296,151 @@ The following sequence diagram shows how the flow of the example execution:
+
+The `EditPersonCommandParser` class will first obtain a clone of the default `Person` object via the `getDefault` method, and then update the fields of the `Person` object with the results of parsing flags in the previous stage when the corresponding flags are found in the user input.
+
+The following is an example of how it can be used:
+```
+Person person = Person.getDefaultPerson();
+            person.setNameIfNotDefault(parseField("name", args, Name::of, nameIsOptional));
+            person.setPhoneIfNotDefault(parseField("phone", args, Phone::of));
+            person.setEmailIfNotDefault(parseField("email", args, Email::of));
+            person.setAddressIfNotDefault(parseField("address", args, Address::of));
+            person.setSubjectsIfNotDefault(parseField("subject", args, Subjects::of));
+            person.setTagsIfNotDefault(parseField("tag", args, Tags::of));
+            person.setRemarkIfNotDefault(parseField("remark", args, Remark::of));
+```
+##### 4. EditPersonCommand Implementation
+The `EditPersonCommand` class will execute the command and update the model accordingly.
+
+Realising the commonality between all kinds of edit commands, we have created an abstract generic class, `EditCommand`, to serve as template class containing the logic of editing a `ListEntry` object.
+
+The signature of the `EditCommand` class is as follows:
+```
+public abstract class AbstractEditCommand<T extends ListEntry<? extends T>> extends Command
+```
+And the concrete edit classes (`EditPersonCommand` class for example) can be simply implemented by "filling in the blank" and providing the correct parameters to the template class.
+
+The template for the edit process is as follows:
+```
+initModelMethods();
+init();
+editFields();
+updatePersonLessonMap();
+validateEditedAndWriteBack();
+showMethod.accept(edited);
+return new CommandResult("Edit success.\n from: " + original.toString() + "\n to: " + edited.toString());
+```
+The `EditPersonCommand` class can be simply implemented by overriding the abstract methods in the `EditCommand` class.
+
+The most important method to be overridden is the initModelMethods(), which looks like this in the `EditPersonCommand` class:
+```
+ @Override
+    protected void initModelMethods() {
+        currentShownEntry = model.getCurrentlyDisplayedPerson();
+        list = model.getFilteredPersonList();
+        hasClashWith = model::hasPersonClashWith;
+        deleteMethod = model::deletePersonForEdit;
+        addMethod = model::addPerson;
+        getClashingEntry = model::getPersonClashWith;
+        showMethod = model::showPerson;
+    }
+```
+
+#### Design considerations:
+
+**Aspect: How to keep the code clean and maintainable while providing the flexibility and variety of valid user input while .
```

Comment:
I think the last while is extra

Reply:
Thank you for pointing that out. Fixed in the latest commit.